diff --git a/CRM/Contact/BAO/Relationship.php b/CRM/Contact/BAO/Relationship.php
index c4aa4de85e1..5a91dc32106 100644
--- a/CRM/Contact/BAO/Relationship.php
+++ b/CRM/Contact/BAO/Relationship.php
@@ -9,6 +9,8 @@
  +--------------------------------------------------------------------+
  */
 
+use Civi\Api4\Contact;
+
 /**
  * Class CRM_Contact_BAO_Relationship.
  */
@@ -1990,6 +1992,22 @@ public static function membershipTypeToRelationshipTypes(&$params, $direction =
     }
   }
 
+  /**
+   * Format the contact name to show on the relationship
+   * Currently adds "deceased" like on contact summary if applicable.
+   *
+   * @param array $contact
+   *
+   * @return string
+   */
+  public static function formatContactName(array $contact): string {
+    $name = $contact['name'];
+    if (!empty($contact['is_deceased'])) {
+      $name .= ' <span class="crm-contact-deceased">(' . ts('deceased') . ')</span>';
+    }
+    return $name;
+  }
+
   /**
    * Wrapper for contact relationship selector.
    *
@@ -2048,6 +2066,13 @@ public static function getContactRelationshipSelector(&$params) {
 
       $displayName = CRM_Contact_BAO_Contact::displayName($params['contact_id']);
 
+      $relationshipContactIDs = CRM_Utils_Array::collect('cid', $relationships);
+      $contactIsDeceased = Contact::get(FALSE)
+        ->addWhere('id', 'IN', $relationshipContactIDs)
+        ->addSelect('is_deceased')
+        ->execute()
+        ->indexBy('id');
+
       // format params
       foreach ($relationships as $relationshipId => $values) {
         $relationship = [];
@@ -2068,6 +2093,11 @@ public static function getContactRelationshipSelector(&$params) {
           FALSE,
           $values['cid']
         );
+
+        $values['name'] = self::formatContactName([
+          'name' => $values['name'],
+          'is_deceased' => $contactIsDeceased[$values['cid']]['is_deceased'],
+        ]);
         $relationship['sort_name'] = $icon . ' ' . CRM_Utils_System::href(
             $values['name'],
             'civicrm/contact/view',
diff --git a/CRM/Contact/Page/View/Relationship.php b/CRM/Contact/Page/View/Relationship.php
index 26cb59f04ce..a740fa322fe 100644
--- a/CRM/Contact/Page/View/Relationship.php
+++ b/CRM/Contact/Page/View/Relationship.php
@@ -9,6 +9,8 @@
  +--------------------------------------------------------------------+
  */
 
+use Civi\Api4\Contact;
+
 /**
  *
  * @package CRM
@@ -68,6 +70,15 @@ public function view() {
     $relationship = new CRM_Contact_DAO_Relationship();
     $relationship->id = $viewRelationship[$this->getEntityId()]['id'];
 
+    $viewRelationship[$this->getEntityId()]['name'] = CRM_Contact_BAO_Relationship::formatContactName([
+      'name' => $viewRelationship[$this->getEntityId()]['name'],
+      'is_deceased' => Contact::get(FALSE)
+        ->addWhere('id', '=', $viewRelationship[$this->getEntityId()]['cid'])
+        ->addSelect('is_deceased')
+        ->execute()
+        ->first()['is_deceased'],
+    ]);
+
     if ($relationship->find(TRUE)) {
       if (($viewRelationship[$this->getEntityId()]['rtype'] == 'a_b') && ($this->getContactId() == $relationship->contact_id_a)) {
         $this->assign("is_contact_id_a", TRUE);
